hrjobcontract/js/src/job-contract/controllers/modal/modal-contract-new.controller.js

```diff
@@ -381,6 +412,35 @@ define([
           $rootScope.options.pay[optionName] = _.mapValues(_.keyBy(data, 'value'), 'label');
         });
     }
+
+    /**
+     * Loads updated record of providers option group
+     *
+     * @param {Array} providers
+     * @returns {Promise}
+     */
+    function loadProviderOptions (providers) {
+      var providersList = [];
+      var optionGroup = '';
+
+      return utilsService.getOptionValues(providers)
+        .then(function (data) {
+          providers.forEach(function (provider) {
+            optionGroup = provider.substr(5);
```

Comment:
this looks too much ad-hoc. Why `5`? What are we extracting here? could you move this to a utility function so that both the name and the documentation would make it clearer?

Reply:
The provider option group is of the format `hrjc_*` (hrjc_health_insurance_provider). i was removing the `hrjc_` part so i can use the remaining part (health_insurance_provider for instance). I wanted to avoid declaring the provider as object initially. I will refactor to make it cleaner. ♻️